Competition2018/src/main/java/competition/subsystems/autonomous/AutonomousDecisionSystem.java

```diff
@@ -7,107 +7,156 @@
 import xbot.common.command.BaseSubsystem;
 import xbot.common.math.ContiguousHeading;
 import xbot.common.math.FieldPose;
 import xbot.common.math.XYPair;
+import xbot.common.properties.BooleanProperty;
+import xbot.common.properties.XPropertyManager;
 
 @Singleton
 public class AutonomousDecisionSystem extends BaseSubsystem {
 
-    GameDataSource gameData;
+    final GameDataSource gameData;
+    final BooleanProperty robotOnRight;
     
     @Inject
-    public AutonomousDecisionSystem(GameDataSource gameData) {
+    public AutonomousDecisionSystem(GameDataSource gameData, XPropertyManager propMan) {
         this.gameData = gameData;
+        robotOnRight = propMan.createPersistentProperty(getPrefix() + "Robot On Right", true);
+    }
+    
+    public void setRobotPosition(boolean onRight) {
+        robotOnRight.set(onRight);
     }
     
     public Supplier<List<FieldPose>> getAutoPathToFeature(GameFeature feature) {
-        return () -> chooseBestPathToFeature(feature);
+        return () -> chooseBestPathToFeature(feature, robotOnRight.get());
     }
     
-    private List<FieldPose> chooseBestPathToFeature(GameFeature feature) {
-        OwnedSide targetSide = gameData.getOwnedSide(feature);
+    private List<FieldPose> mirrorPath(List<FieldPose> path) {
+        List<FieldPose> flippedPath = new ArrayList<FieldPose>();
+        
+        for (FieldPose pose : path) {
+            flippedPath.add(new FieldPose(
+                    new XYPair(-pose.getPoint().x, pose.getPoint().y),
+                    new ContiguousHeading(pose.getHeading().getValue())
+                )
+            );
+        }
         
+        return flippedPath;
+    }
+    
+    private List<FieldPose> chooseBestPathToFeature(GameFeature feature, boolean onRight) {
+        OwnedSide targetSide = gameData.getOwnedSide(feature);
+        log.info("Target Side is: " + targetSide);
+        List<FieldPose> bestPath = createPathToNowhere();
         switch (feature) {
         case SWITCH_NEAR:
             switch (targetSide) {
             case LEFT:
-                log.info("Creating path to Left Switch");
-                return createPathToLeftSwitch();
+                log.info("Creating path to Left Switch Plate");
+                if (onRight) {
+                    bestPath = createPathToDistantSwitchPlate();
+                } else {
+                    bestPath = mirrorPath(createPathToNearbySwitchPlate());
+                }                
+                break;
             case RIGHT:
-                log.info("Creating path to Right Switch");
-                return createPathToRightSwitch();
+                log.info("Creating path to Right Switch Plate");
+                if (onRight) {
+                    bestPath = createPathToNearbySwitchPlate();
+                } else {
+                    bestPath = mirrorPath(createPathToDistantSwitchPlate());
+                }  
+                break;
             case UNKNOWN:
```

Comment:
If/else might have been better than switch given how much repetition we have around default/failure cases.